Read parquet using DuckDB parquet reader

ducndh opened the pull request:
Fixing metadata mismatch when calling table function with GPU to support read_parquet from inside gpu_processing.

DuckDB only loads the necessary columns from the parquet file when doing a read_parquet. Therefore, a different column projection is treated as a different table in the DuckDB metadata. As a result, we have to use the file path + column names necessary for that projection for the table cache.

ducndh committed:
Finish fixing the parquet with unoptimized read through duckDB

ducndh committed:
clean code with TPC-H Parquet GPU Command for test

ducndh committed:
remove duplicate file

bwyogatama commented:
This is great! Thanks for doing this. 
1. Can you add scripts or instructions on how to generate the parquet file so that people can try right away?
2. Can you add some basic test suites for the Parquet support? Here is some example: https://github.com/sirius-db/sirius/blob/dev/test/sql/tpch-sirius.test

ducndh committed:
feat: Update documentation and add new TPC-H Parquet GPU tests

This commit includes:
- Updates to  for improved clarity.
- Addition of  for parquet generation.
- Addition of  for TPC-H Parquet GPU testing.

These changes are in response to feedback and updates for the existing pull request.

ducndh commented:
I just added the test files and the script to generate Parquet file that is consistent with the way we generated the TPCH dataset. Running "make test" matches all the solutions provided in the tpch-dbgen.

bwyogatama commented:
Hi Duke, I would like to merge this PR, can we rebase to the dev branch before merging?

felipeblazing commented on src/gpu_buffer_manager.cpp:
```diff
@@ -423,36 +430,45 @@ GPUBufferManager::createTableAndColumnInGPU(Catalog& catalog, ClientContext& con
-    //convert column_name to uppercase
+
+    // Normalize names
+    string up_table_name = table_name;
```
is there a reason why this case handling isn't done before it gets to the Sirius part of the code? Duckdb treats these as all case insensitive  can we rely on duckdb handling this?

felipeblazing commented on src/operator/gpu_physical_table_scan.cpp:
```diff
@@ -73,8 +73,76 @@ GPUPhysicalTableScan::GPUPhysicalTableScan(vector<LogicalType> types, TableFunct
+        projected_names.push_back(names[col_id.GetPrimaryIndex()]);
+    }
+    // Sort for consistent hashing across different query orders
+    sort(projected_names.begin(), projected_names.end());
```
if the columns aren't in the same order this isn't really the same. What is the situation this is trying to handle?
